src/main/scala/dpla/ingestion3/mappers/providers/Extractor.scala

```diff
@@ -9,4 +10,14 @@ import dpla.ingestion3.model.{DplaMapData, EdmAgent}
 trait Extractor {
   def build(rawData: String): DplaMapData
   def agent: EdmAgent
-}
+  /**
+    * Generates an MD5 checksum from the given providerId
+    * Implemented per provider
+    *
+    * DigestUtils.md5Hex(id)
+    *
+    * @return MD5 hash value
+    * @throws NotImplemented
+    */
+  def mintItemId(rawData: Any): String
```

Comment:
@mdellabitta The issue I see with that is that we will then need to repeat the ID selection code when minting the `id` value and the `uri` and that leaves us vulnerable to not updating how the id is selected in all places.